### PR TITLE
feat: adding advanced api implementation

### PR DESCRIPTION
**Pull request by ScriptedAlchemy**
https://github.com/webpack/webpack/pull/10960

### CODE REVIEW

**Comment by zhangwilling** [comment]
cool thing ! i got this news from your twitter.

File: advanced-api/automatic-vendor-sharing/app1/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      remotes: {
+        app2: "app2",
+      },
+      exposes: {
+        "./Button": "./src/Button",
+      },
+      shared: {
+        ...deps,
+        react: {
+          singleton: true,
```
Comment (yordis) [inline advanced-api/automatic-vendor-sharing/app1/webpack.config.js:52]: Is this new to the API now?

File: advanced-api/dynamic-remotes/app1/src/App.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
+import React from "react";
+
+function loadComponent(scope, module) {
+  return async () => {
+    // Initializes the share scope. This fills it with known provided modules from this build and all remotes
+    await __webpack_init_sharing__("default");
```
Comment (yordis) [inline advanced-api/dynamic-remotes/app1/src/App.js:6]: Where can I read about this?

File: advanced-api/dynamic-remotes/app1/src/App.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    const container = window[scope]; // or get the container somewhere else
+    // Initialize the container, it may provide shared modules
+    await container.init(__webpack_share_scopes__.default);
```
Comment (yordis) [inline advanced-api/dynamic-remotes/app1/src/App.js:9]: Where can I read about this?

File: advanced-api/automatic-vendor-sharing/app1/webpack.config.js (hunk 1 above)
Comment (ScriptedAlchemy) [reply advanced-api/automatic-vendor-sharing/app1/webpack.config.js:52]: yep, all things un shares is new to the webpack api. beta.17 is unreleased but will be in a few days

File: advanced-api/dynamic-remotes/app1/src/App.js (hunk 2 above)
Comment (ScriptedAlchemy) [reply advanced-api/dynamic-remotes/app1/src/App.js:6]: Source code :P

File: advanced-api/dynamic-remotes/app1/src/App.js (hunk 3 above)
Comment (ScriptedAlchemy) [reply advanced-api/dynamic-remotes/app1/src/App.js:9]: source code, or the webpack PR that was opened

**Comment by ScriptedAlchemy** [comment]
@yordis https://github.com/webpack/webpack/pull/10960

**Pull request merged**